### ISSUE TITLE
Segmentation fault in Aravis when launching driver with the scanCONTROL 3060-25

### ISSUE THREAD

**Issue by dave992** [sam-xl/scancontrol#21]
The driver seems to fail in Aravis (dependency of the SDK) with a segmentation fault (most of the time), when used with the Micro-Epsilon scanCONTROL 3060-25:
```
#0  0x00007ffff3625eb5 in arv_stream_set_emit_signals () at /usr/local/lib/libaravis-0.6.so.0
#1  0x00007ffff4722d90 in transfer_profiles (hllt=0x555555853b20, transfer_profile_type=(unknown: 543583329), is_active=0) at ../src/mescan_adv.c:2111
#2  0x00007ffff4720594 in disconnect_llt (hllt=0x555555853b20) at ../src/mescan_adv.c:724
#3  0x00007ffff4720514 in connect_llt (hllt=0x555555853b20) at ../src/mescan_adv.c:699
#4  0x00007ffff6fdfd61 in CInterfaceLLT::Connect() (this=0x7fffffffc7f0) at ../src/llt.cpp:252
#5  0x00005555555d70e9 in scancontrol_driver::ScanControlDriver::ScanControlDriver(ros::NodeHandle, ros::NodeHandle) (this=0x7fffffffc540, nh=..., private_nh=...) at /home/davekroezen/workspaces/scancontrol_ws/src/scancontrol/micro_epsilon_scancontrol_driver/src/driver.cpp:117
#6  0x00005555555d0932 in main(int, char**) (argc=1, argv=0x7fffffffc9b8) at /home/davekroezen/workspaces/scancontrol_ws/src/scancontrol/micro_epsilon_scancontrol_driver/src/node.cpp:13

```

**Comment by dave992** [comment]
I am still unsure of what changed and caused this behavior. I am still using the same version of Aravis (0.6.4), the SDK did not have any API changes moving from 0.2.0 to 0.2.4. and the older 2950-25 did work without issue before. 

The examples from the SDK still work fine, also with the 3060-25 sensor on which the above issue was observed. The most notable difference between the SDK examples and the driver is the use of a pointer to the device interface by the SDK examples. 

Changing the driver to also use a pointer to the device interface seems to have solved the issue. @gavanderhoorn @IKapitaniuk  any idea what can be the reason this fixes the issue?

**Comment by IKapitaniuk** [comment]
@dave992 Just to check that the problem can be in the inconsistent state of CInterfaceLLT object could you test any of two cases for the old code?

1) Comment the line:
```
        device_interface = CInterfaceLLT();
```
to avoid invoking the assignment operator + ctor and dtor of the temporary object. Anyway, this line is excessive since, by that line, the object was already initialized by the default ctor. 

or 

2) Explicitly initialize the object using the initializing list, i.e.
```
ScanControlDriver::ScanControlDriver(ros::NodeHandle nh, ros::NodeHandle private_nh):  device_interface(CInterfaceLLT())
{
...
}
```
Here the ctor will be called only once without any extra steps.

Both cases do the same, choose what you prefer.

**Comment by dave992** [comment]
I went with 1, which also solved the issue. Safe to assume that there is indeed an inconsistent state of the `CInterfaceLLT` object when copied from the temporary object. 

> Anyway, this line is excessive since, by that line, the object was already initialized by the default ctor.

Yes indeed, easy to miss these mistakes when you write and 'review' it yourself. I will need to have a look at the rest of the driver to see if I made this or comparable mistakes more often. Thanks for the input!
